src/consumer/component/consumer-component.js

```diff
@@ -641,14 +634,20 @@ export default class Component {
         specs = path.join(bitDir, bitJson.getSpecBasename());
       }
     }
-    const distPath = path.join(process.cwd(), bitJson.distEntry);
-    const entry = fs.existsSync(distPath) ? distPath : process.cwd();
+    const enrtyPath = path.join(process.cwd(), bitJson.distEntry);
+    const cwd = fs.existsSync(enrtyPath) ? enrtyPath : process.cwd();
+
     const files = componentMap.files;
-    const vinylFiles = Object.keys(files).map(file => new SourceFile(vinylFile.readSync(path.join(consumerPath, files[file]), { base: entry })));
+    const vinylFiles = Object.keys(files).map((file) => {
+      const sourceFile = new SourceFile(vinylFile.readSync(path.join(consumerPath, files[file]), { base: cwd }));
```

Comment:
This logic should be inside SourceFile class in the load function.
The load should get a path and base and call the vinylFile.readSync
And probably also define the distFilePath

Also please change the name of files.js to SourceFile.js